### PR TITLE
Allow search_in only in specific fields

### PR DESCRIPTION
**Pull request by kevindew**
This specifies which fields can be searched in for GetContentCollection
as root level fields and nested fields. These are limited since using
search_in on a field type that doesn't support `ilike` will result in a
500 error.

/cc @rubenarakelyan

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -121,6 +121,8 @@
     t.datetime "created_at",                    null: false
     t.datetime "updated_at",                    null: false
     t.integer  "position",          default: 0, null: false
+    t.integer  "content_item_id"
```
Comment (rubenarakelyan) [inline db/schema.rb]: Is this a stray schema change?

File: app/queries/get_content_collection.rb [hunk 2]
```diff
@@ -82,19 +82,33 @@ def default_search_fields
       presenter::DEFAULT_SEARCH_FIELDS
     end
 
+    def allowed_search_fields
+      presenter::SEARCH_FIELDS
+    end
+
+    def nested_search_fields
```
Comment (rubenarakelyan) [inline app/queries/get_content_collection.rb]: Would it be clearer to call this `allowed_nested_search_fields` or similar?

File: app/queries/get_content_collection.rb [hunk 3]
```diff
@@ -82,19 +82,33 @@ def default_search_fields
...
+      raise_error("Invalid search field: #{field}") unless valid_search_field(field)
+      # special case as description is a JSON object that just contains a hash
+      # of value
+      return "description->>'value'" if field == "description"
+      return field unless field.include?(".")
+      elements = field.split(".")
+      "#{elements[0]}->>'#{escape_nested_field(elements[1])}'"
+    end
+
+    def valid_search_field(field)
```
Comment (dougdroper) [inline app/queries/get_content_collection.rb]: `valid_search_field?`

File: db/schema.rb (hunk 1 above)
Comment (kevindew) [inline db/schema.rb]: argh! I only made this mistake the other day. Thanks for catching

**Comment by kevindew** [comment]
Ok, comments and mess up addressed.

**Comment by tijmenb** [comment]
👍 perhaps an update to the docs would be good as well.

**Comment by kevindew** [comment]
@tijmenb d'oh - of course. Does that read ok to you?

**Pull request merged**